src/java.base/unix/native/libjli/java_md_common.c

```diff
@@ -128,8 +128,12 @@ Resolve(char *indir, char *cmd)
 {
     char name[PATH_MAX + 2], *real;
 
-    if ((JLI_StrLen(indir) + JLI_StrLen(cmd) + 1)  > PATH_MAX) return 0;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
+    if ((JLI_StrLen(indir) + JLI_StrLen(cmd) + 2) > sizeof(name)) return 0;
     JLI_Snprintf(name, sizeof(name), "%s%c%s", indir, FILE_SEPARATOR, cmd);
+#pragma GCC diagnostic pop
```

Comment:
Wouldn't it be better to just call JLI_Snprintf without the precheck and check the result to determine whether it was successful or was truncated?  I think that kind of check is supposed to make gcc's truncation checker happy.

Reply:
The warning has gone when using return value from `JLI_Snprintf()` in new commit!